### DISCUSSION TITLE
Hosting CoreWCF Service in WinForms-App or WPF-App?

### DISCUSSION THREAD

**Discussion by GoldyWang2013** [CoreWCF/CoreWCF#986] [Q&A]
Can hosting CoreWCF Service on WindowsForm-App or WPF-App?
I tried it , but failed...
can somebody tell me, how to do it..

**Comment by pkar70** [comment]
I have same question - can CoreWCF be used in WPF app?
Both as server, and a client.
App has to be WPF (maybe WinUi3), and should be able to communicate with same app running on another desktop. UI is very important, as it is core functionality of app.
